api-testing/multicloud-api/src/main/java/com/soda/utils/BinaryUtils.java

```diff
@@ -0,0 +1,223 @@
+/*
+  Copyright 2020 The SODA Authors.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+package com.soda.utils;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Locale;
+
+/**
+ * Utilities for encoding and decoding binary data to and from different forms.
+ */
+public class BinaryUtils {
+    static final int ONE_MB = 1024 * 1024;
+
+    /**
+     * LINK: https://docs.aws.amazon.com/AmazonS3/latest/API/samples/AWSS3SigV4JavaSamples.zip.
+     * Converts byte data to a Hex-encoded string.
+     *
+     * @param data
+     *            data to hex encode.
+     *
+     * @return hex-encoded string.
+     */
+    public static String toHex(byte[] data) {
```

Comment:
Please Update the licence accordingly if you are referring code from other source.

1 query . Is it possible to use the already available library functions or Do we really need to write these methods for our own
[https://github.com/aws/aws-sdk-java/blob/master/aws-java-sdk-core/src/main/java/com/amazonaws/util/BinaryUtils.java](url)

Reply:
Yes, it is possible but why we are increase or code size using whole library we need only piece of code, Another problem is we have same lib `uk.co.lucasweb`  classes will be conflicts with this.

Reply:
Which licence i will be use.

Reply:
Hi @Click2cloud-Wadhai, I think what @himanshuvar means is if we referred any code from other places, we need to keep their license in the code.

Reply:
I have added licence in the code please review.

Reply:
https://github.com/Click2cloud-Wadhai As we discussed earlier too, any code reference should have proper credit, whether it is license or reference. Thanks